### PR TITLE
Add com.eduke32.EDuke32.DukeNukem3DShareware

### PR DESCRIPTION
**Pull request by FakeShemp**
With my MR of EDuke32 (https://github.com/flathub/flathub/pull/600) brought up to some kind of standard I tried making an addon install of the shareware Duke Nukem 3D. However, I don't know how to do an extension with extra-data, so this is a WIP. Any help would be appreciated.

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build com.eduke32.EDuke32.Plugin.DukeNukem3DShareware

File: com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+  <id>com.eduke32.EDuke32.Plugin.DukeNukem3DShareware</id>
+  <extends>com.eduke32.EDuke32.desktop</extends>
```
Comment (bilelmoussaoui) [inline com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.appdata.xml]:
```suggestion
  <extends>com.eduke32.EDuke32</extends>
```
You should use the app-id and not the launchable tag here

File: com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+id: com.eduke32.EDuke32.Plugin.DukeNukem3DShareware
+branch: 1-3.32
+runtime: com.eduke32.EDuke32
+runtime-version: stable
```
Comment (bilelmoussaoui) [inline com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.yaml]:
```suggestion

```

File: com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+id: com.eduke32.EDuke32.Plugin.DukeNukem3DShareware
+branch: 1-3.32
```
Comment (bilelmoussaoui) [inline com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.yaml]:
```suggestion
```

File: com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+runtime: com.eduke32.EDuke32
+runtime-version: stable
+sdk: org.gnome.Sdk//3.32
```
Comment (bilelmoussaoui) [inline com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.yaml]:
```suggestion
sdk: org.gnome.Sdk
runtime-version: "3.32"
```
Please split this into sdk & runtime-version.

**Comment by FakeShemp** [comment]
@bilelmoussaoui I've used TingPing's [guide](https://blog.tingping.se/2018/03/18/flatpaking-plugins.html) and it doesn't follow your style. Have there been updates to the format in writing plugins?

**Comment by bilelmoussaoui** [comment]
The guide is a bit old(uses 3.26 while we are close to 3.34.). Please follow the one in Flathub wiki.

**Comment by FakeShemp** [comment]
@bilelmoussaoui I can't find that one in the wiki. Could you link it to me, please.

**Comment by LeandroStanger** [comment]
bot, build com.eduke32.EDuke32.Plugin.DukeNukem3DShareware

File: com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.yaml (hunk 4 above)
Comment (gasinvein) [reply com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.yaml]: The original one was correct, your suggestion is not. It's an extension, so runtime here is the app, not Gnome runtime.

Comment (bilelmoussaoui) [reply com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.yaml]: @gasinvein Oh my bad! I just saw this.  @FakeShemp Can you please revert this, please? Sorry for this.

Comment (gasinvein) [reply com.eduke32.EDuke32.Plugin.DukeNukem3DShareware.yaml]: @bilelmoussaoui That's alright, it's already replaced.

**Comment by nedrichards** [comment]
Extra-data extensions are only supported in 1.0.9, 1.2.5 and 1.4.2 or later. You need to use required-flatpak to make sure this isn't installed on lower versions.

**Comment by LeandroStanger** [comment]
bot, build com.eduke32.EDuke32.Plugin.DukeNukem3DShareware

**Comment by FakeShemp** [comment]
bot, build com.eduke32.EDuke32.DukeNukem3DShareware

**Comment by FakeShemp** [comment]
bot, build com.eduke32.EDuke32.Plugin.DukeNukem3DShareware

**Comment by bilelmoussaoui** [comment]
@FakeShemp it seems that flathub bot is not doing its job.

**Comment by FakeShemp** [comment]
bot, build com.eduke32.EDuke32.DukeNukem3DShareware

**Comment by FakeShemp** [comment]
bot, build com.eduke32.EDuke32.DukeNukem3DShareware

**Comment by FakeShemp** [comment]
bot, build com.eduke32.EDuke32.DukeNukem3DShareware

**Comment by FakeShemp** [comment]
bot, build com.eduke32.EDuke32.DukeNukem3DShareware

**Comment by gasinvein** [comment]
Why build this as extension btw? Putting the extra-data source into the app bundle would be easier.

**Comment by FakeShemp** [comment]
Because some people maybe don't want it and prefer to only play the full game or NAM or something.

**Pull request closed without merging**